bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -907,21 +941,40 @@ public void stop() {
 }
 
 @Override
-public boolean setText(String html, boolean trusted) {
-	char[] data = new char[html.length() + 1];
-	html.getChars(0, html.length(), data, 0);
-	return webView.NavigateToString(data) == COM.S_OK;
+public URI getUriForCustomText() {
+	return URI.create(ABOUT_BLANK);
+}
+
+private boolean isLocationForCustomText(String location) {
+	return URI_FOR_CUSTOM_TEXT_PAGE.equals(URI.create(location));
```

Comment:
Is there a guarantee that `URI.create()` succeedson the location? I am concerned that this may result in problems like #1455

So maybe we could also do a simple string comparison?

Reply:
I had introduced uri comparision cause the path can have different notations
like // and / or /../../ and many more. URI comparision seems to handle that. but yes uri always needs a protocol in the beginning (file://, html://, etc) and the error that I saw in #1455 seems to occur when the location starts like /something/....

But i am pretty sure string comparision might be a bit difficult. we need a better way to check this.

Reply:
YOu should normalize the URI and catch exception, if its not a parseable URI just return false.

Reply:
Okay maybe one stupid question: what to do when the protocol is missing. In the Unix filesystem, it could be that we receive paths startign with "/"
One solution would to always prepend html:// or file:// if a protocol is missing the way its done in in setWebPage:

`if (!url.matches("[a-z][a-z0-9+.-]*:.*")) {
		url = "http://" + url;
	}`
	
but again for checking the path for custom text (in isLocationForCustomText) it makes more sense to prepend a file:// instead of html://
Regarding normalizing, URI seems to take care of that implicitly, the only problem I forsee is when the protocol is missing.

Reply:
The API claims that the location is always a valid URI, so anything passed in here that is not an URI must return false.

Reply:
> One solution could be to always prepend html:// or file:// if the protocol is missing, just the way its done in setWebPageData:
> 
> `if (!url.matches("[a-z][a-z0-9+.-]*:.*")) { url = "http://" + url; }`
> 
> but again, for checking the path for custom text (in isLocationForCustomText) it makes more sense to prepend a file:// instead of html://

Why is there any need to modify the location string? Currently, this is only produced internally by retrieving the current browser URL. And this should always have a protocol prefix (or be some magic string like "about:blank", shouldn't it? When I print out the locations passed to method in all the existing browser tests, this assumption is fulfilled. Maybe some specific handling for values that are, by intent, no URIs is necessary (like "about:blank")?
In case this method will be exposed at some point in time, you should only pass some URL that you have retrieved from the browser into it as well, which then should also have an according protocol prefix. In case someone passes any other string to the method (without a protocol prefix), I agree with @laeubi that this should simply return false. Is there anything speaking against this?

Reply:
Alright then, I'll just handle the exception and return false here.

Reply:
Looks sound to me. @laeubi does it reflect your proposal/expectation now?